.github/workflows/acceptance_tests.yml

```diff
@@ -0,0 +1,78 @@
+name: Acceptance tests
+
+on:
+  pull_request:
+    types:
+    - synchronize
+
+jobs:
+  acceptance-tests:
+    strategy:
+      max-parallel: 1
+      matrix:
+        include:
+          - image_name: py
```

Comment:
I think you need to build this one first outside the matrix and make it a dependency for the other steps. The others can then execute in parallel.

Reply:
There is a way to use caching with an action called [docker-build-push](https://github.com/docker/build-push-action), but I could not get it to work in this case. It gets tricky when you have separate jobs but you wish to share some artifacts (i.e. the renku/py image) between them. I will look into this in the future however - I think the biggest problem was because of a bug in the action related to not recognizing the build--args I was passing in.